### PR TITLE
Enable codecov-io for code coverage reporting in GitHub

### PR DESCRIPTION
**Pull request by meltsufin**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -39,6 +39,10 @@ script:
       mvn clean install -B -U
     fi
 
+# enable codecov-io
+after_success:
+  - bash <(curl -s https://codecov.io/bash)
```
Comment (patflynn) [inline .travis.yml]: this seems.. unsafe.  Is there no direct integration we can provide? Does it have to be through Travis?

Comment (meltsufin) [inline .travis.yml]: This seems to be the recommended way to integrate it. We do have to trust them just like we trust Travis.

Comment (meltsufin) [inline .travis.yml]: I will enable it only for pull requests that don't expose secrets.

Comment (meltsufin) [inline .travis.yml]: Actually, it's not that straightforward. Pull requests from committers have access to secrets. I need to investigate this a bit more to find a workable solution.

**Comment by meltsufin** [comment]
@patflynn Can you take a look and see if protecting the secrets the way I did is good?

File: .travis.yml [hunk 2]
```diff
@@ -39,6 +39,16 @@ script:
       mvn clean install -B -U
     fi
 
+after_success:
+  # hide sensitive data from codecov-io
+  - rm -f config/travis-app-maven-plugin-e1bb638f290e.json
+  - export SONATYPE_USERNAME=[secure]
```
Comment (patflynn) [inline .travis.yml]: why '[secure]'? I know that's the string that travis prints out when it assigns a secret value, but in our case we want to wipe it out.

Comment (meltsufin) [inline .travis.yml]: You're right, doing `unset` on them is cleaner.

**Comment by patflynn** [comment]
LGTM

**Pull request merged**